src/main/java/org/prebid/server/bidder/readpeak/ReadPeakBidder.java

```diff
@@ -0,0 +1,253 @@
+                    .id(StringUtils.isNotBlank(extImpReadPeak.getSiteId())
+                            ? extImpReadPeak.getSiteId() : request.getSite().getId())
+                    .publisher(publisher)
+                    .build();
+            requestBuilder.site(site);
+        } else if (request.getApp() != null) {
+            final App app = request.getApp().toBuilder()
+                    .id(StringUtils.isNotBlank(extImpReadPeak.getSiteId())
+                            ? extImpReadPeak.getSiteId() : request.getApp().getId())
+                    .publisher(publisher)
+                    .build();
+            requestBuilder.app(app);
+        }
+
+        final BidRequest outgoingRequest = requestBuilder.build();
+
+        return BidderUtil.defaultRequest(outgoingRequest, endpointUrl, mapper);
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            final List<BidderBid> bids = extractBids(bidResponse);
+            return Result.withValues(bids);
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidResponse bidResponse) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidResponse);
+    }
+
+    private List<BidderBid> bidsFromResponse(BidResponse bidResponse) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(List::stream)
+                .map(bid -> makeBid(bid, bidResponse))
+                .filter(Objects::nonNull)
+                .toList();
+    }
+
+    private BidderBid makeBid(Bid bid, BidResponse bidResponse) {
+        if (bid.getExt() == null) {
+            throw new PreBidException("Bid ext is null");
+        }
+        final Bid resolvedBid = resolveMacros(bid);
+        if (resolvedBid.getExt() == null) {
+            throw new PreBidException("Bid ext is null");
+        }
+        final BidType bidType = getBidType(resolvedBid);
+
+        // Parse existing ExtBidPrebidMeta
+        final ExtBidPrebid extBidPrebid = parseExtBidPrebidMeta(resolvedBid);
+        final ExtBidPrebidMeta meta = extBidPrebid.getMeta();
+
+        // Validate rendererName and rendererVersion
+        if (StringUtils.isBlank(meta.getRendererName())) {
+            throw new PreBidException("RendererName should not be empty");
+        }
+        if (StringUtils.isBlank(meta.getRendererVersion())) {
+            throw new PreBidException("RendererVersion should not be empty");
+        }
```

Comment:
is it just a copy from another bidder? this logic is irrelevant to the GO version, please pay attention to that